Convert queries to promises, useMasterKey

nbw opened the pull request:
Convert now-unsupported parse _find_, _first_ methods to follow promises styling.

Also, convert `Parse.Cloud.useMasterKey()` to new styling of `{ useMasterKey: true }`. 

- [migration guide link](https://www.appcoda.com/parse-migration-part3/)
- [parse's promises documentation](http://parseplatform.github.io/docs/js/guide/#promises)

Commit:
Convert queries to promises, useMasterKey() method => { useMasterKey: true }

rselk commented on cloud/Stats.js:
```diff
@@ -53,8 +53,6 @@ function sumArray(arr) {
 Parse.Cloud.define("stats", function(request, response) {
 
-	Parse.Cloud.useMasterKey();
-
```
Why was this removed? Just wondering.

nbw replied:
refer to https://www.appcoda.com/parse-migration-part3/

rselk replied:
👍  Thanks! That makes sense.

Commit:
Convert find/first methods to promise formatting (part 2)

- convert rest of find/first methods to promise format
- useMasterKey() to { useMasterKey: true } formatting

Commit:
remove extraneous method argument

Commit:
/parse on server url

Commit:
fixed the SERVER_URL on heroku

Commit:
Convert find method to promise styling

- missing find that hadn't been coverted

Commit:
Commenting out push notifications for startDriving & stopDriving endpoints

Commit:
removed extraneous masterKey input arg

Commit:
Convert Save method to promise format

* refer to: http://parseplatform.github.io/docs/js/guide/#promises

Commit:
Convert fetch to promise format (plus two missing save method calls)

* refer to http://parseplatform.github.io/docs/js/guide/#promises

Commit:
Convert destroyAll method to promise format

- refer to: http://parseplatform.github.io/docs/js/guide/#promises

Commit:
Convert Get method to promise format

* refer to: http://parseplatform.github.io/docs/js/guide/#promises

Commit:
FIREBASE CONFIG

- added a field for Firebase Server Key
- added a field for Firebase Sender ID

Commit:
renabled push notifications in Driving.js (removed earlier for debugging)

Commit:
Push notification method upgraded to include master key

Commit:
Commented out iOS push notif for friends request, pushCallbackCounter to 1

Commit:
Replaced two remaining instances of Parse.Cloud.useMasterKey with new format

Commit:
tiny push to force bcrypt update

Commit:
Added masterKey to friend request save(..) method call

Commit:
output Role in afterSave for friend request

Commit:
Inspect Requested User in friend request

Commit:
Missing first param in save method call

Commit:
Updated any save(..) method calls to use the masterKey

* note, first input arg for the save method should be null or empty obj:
     ex: save( {},{ useMasterKey: true } )

Commit:
Enabled VERBOSE logging, commented out Morgon (for now)

Commit:
Update to sendPush method; includes some debug calls

Commit:
Removed debug comments

Commit:
Comment out push notif success logging

Commit:
multiple friend request debugging

Commit:
MasterKey in returned query call

Commit:
Update to verifyPhoneShortCode method: useMasterKey flag

Commit:
POFriendRequest: added masterKey to inner query

Commit:
MasterKey updates, renamed result => res to avoid conflicts

Commit:
serverShortCode debug code

Commit:
sendPhoneShortCode testing

Commit:
sendPhoneShortCode testing

Commit:
Twilio sendSMS => sendMessage

Commit:
Twilio promise format

Commit:
Updated Save method call in POFriendRelation; removed some debug comments

Commit:
masterKey on destroy method calls for Friend Requests and for Friend Relations

Commit:
added a success reponse to sendPush method

Commit:
debugging for multi push notifications

Commit:
Push notif callbackCounter readded

Commit:
Added MasterKey to userTotals find() method call; removed debug comments

Commit:
Mailgun Adapter config
